pageserver/src/object_repository.rs

```diff
@@ -228,7 +231,7 @@ impl Timeline for ObjectTimeline {
     //------------------------------------------------------------------------------
 
     /// Look up given page in the cache.
-    fn get_page_at_lsn(&self, tag: BufferTag, req_lsn: Lsn) -> Result<Bytes> {
+    fn get_page_at_lsn(&self, tag: ObjectTag, req_lsn: Lsn) -> Result<Bytes> {
```

Comment:
Sorry for late reply, Git collapse this comment and I didn't noticed it.
Some for our KV storage we have now enum for keys (ObjectKey) and enum for values (ObjectValue).
Them are defined in different files because used and introduced at different level of abstraction.
May it is not so good and need to fixed. But I do not consider such refactoring as critical tasks now...

ObjectValue enum includes now just 4  variants: Page, WALRecord, RelationSize, Unlink and TimelineMetadata.
I am not sure that last one is needed and it was added by me mostly for easier merging with your code.

Page(Bytes) is used not only for relation pages, but also for all materialize data, i.e. SLRU pages, controlfile.map, two phase files, pg_control...  So it is actually now always "pages" data. So may be wrong term. 
It can be written using put_page_image and extracted get_page_at_lsn or get_page_at_lsn_nowait.
I think that time line metadata can be stored/extract in the same way, this is why I think that TimelineMetadata enum value may be not needed.